backend/controllers/groupDelete.js

```diff
@@ -1,23 +1,33 @@
 const groupDeleteFactory = (getGroupDetailsApi, deleteGroupApi, maintainUrl) => {
-  const stashStateAndRedirectToIndex = (req, res, errors, group) => {
+  const stashStateAndRedirectToIndex = (req, res, errors, group, url) => {
     req.flash('deleteGroupErrors', errors)
     req.flash('group', group)
-    res.redirect(req.originalUrl)
+    res.redirect(url)
   }
 
   const index = async (req, res) => {
     const { group } = req.params
     const hasMaintainAuthUsers = Boolean(res.locals && res.locals.user && res.locals.user.maintainAuthUsers)
-    const groupDetails = await getGroupDetailsApi(res.locals, group)
-    const groupUrl = `${maintainUrl}/${group}`
+    try {
+      const groupDetails = await getGroupDetailsApi(res.locals, group)
+      const groupUrl = `${maintainUrl}/${group}`
 
-    res.render('groupDelete.njk', {
-      group,
-      groupUrl,
-      groupDetails,
-      hasMaintainAuthUsers,
-      maintainUrl,
-    })
+      res.render('groupDelete.njk', {
+        group,
+        groupUrl,
+        groupDetails,
+        hasMaintainAuthUsers,
+        maintainUrl,
+      })
+    } catch (error) {
+      if (error.status === 404) {
+        const groupChildError = [{ href: '#groupCode', text: 'Group does not exist' }]
```

Comment:
variable named incorrectly?  should be `groupError`?

Reply:
Updated (cut and paste error)